### ISSUE TITLE
Freeze the Discriminator: a Simple Baseline for Fine-Tuning GANs

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#28]
# Key Idea

- Propose a very simple, but powerful way of conducting transfer learning with near-SOTA GANs.
- In discriminator model, freeze the lower layers, the feature extractors, and only fine tune higher layers responsible for classification. (Analogous to other transfer learning methods)

# Previous Approaches

### Fine Tuning
- Initialize with pretrained weights and train both G and D for more epochs.
- Susceptible to **over-fitting**.

### Scale/Shift
- Change domain by applying scale/shift as in BN or AdaIN.
- Inferior outcome due to excess restrictions.

### Generative Latent Optimization (GLO)
- Fine-tuning only the Generator using L1 loss and perceptual loss (supervised fashion)
- Blurry images due to L1 loss and absence of adversarial term.

# Formulation

### Unconditional GAN (StyleGAN)

- Freeze lowest 4 layers of Discriminator
- 50K additional iterations.

### Conditional GAN (SNGAN-projection)

- Freeze lowest (3,2,1) layers depending on the dataset. 
- 20K additional iterations.

# Experiment / Result

### Unconditional GAN

- StyleGAN trained on FFHQ => Animal Dataset, Anime Dataset

![image](https://user-images.githubusercontent.com/46308407/104923207-16cc5100-59df-11eb-8849-624d736a136f.png)

The overall semantic is somewhat preserved, as images of corresponding position are from same latent code.

![image](https://user-images.githubusercontent.com/46308407/104923299-35324c80-59df-11eb-9128-6150f07ec89f.png)

- Faces of Anime Characters. Somewhat blurry.

![image](https://user-images.githubusercontent.com/46308407/104923573-90643f00-59df-11eb-8f50-7acbc241274b.png)

### Conditional GAN

- SNGAN-projection trained on ImageNet => Oxford Flower, CUB-200-2011, Caltech-256

![image](https://user-images.githubusercontent.com/46308407/104923394-585cfc00-59df-11eb-942c-04b7df86fdcf.png)

Class consistency better illustrated (right)

![image](https://user-images.githubusercontent.com/46308407/104923448-69a60880-59df-11eb-95b3-919a16de975c.png)

Unstable training for Oxford Flowers, but overall FreezeD performs better.

# Comment

- 아이디어가 너무나 간단. 이론적으로 설명할 부분도 없음. 논문도 굉장히 짧고.
- 하지만 실험 결과가 꽤 좋다 (동물은 확실히 인상적. Anime Dataset의 경우에는 글쎄... Domain 차이가 큰 듯)
- 무엇보다 원래도 GAN training이 만만치 않은데 최근 StyleGAN, BigGAN 등은 모델도 커져서 학습이 굉장히 오래걸리는 단점이 있었음. 이런 것을 극복할 수 있게 해준다는 점에서 실용적인 가치도 충분한 듯.
- 우연히 마주친 첫 국내 대학 논문. 신기했다.
